packages/desktop/components/DappDataBanner.svelte

```diff
@@ -17,11 +20,23 @@
         </Text>
 
         {#if dapp?.metadata?.url}
-            <Text type="sm" textColor="secondary" truncate>
-                {dapp.metadata.url}
-            </Text>
+            <div class="flex flex-row items-center gap-1">
+                {#if verifiedState === DappVerification.Valid}
+                    <Icon name={IconName.ShieldOn} size="xs" textColor="success" />
+                {:else if verifiedState === DappVerification.Invalid || verifiedState === DappVerification.Scam}
+                    <Icon name={IconName.ShieldOff} size="xs" textColor="danger" />
+                {:else if verifiedState === DappVerification.Unknown}
+                    <Icon name={IconName.ShieldOff} size="xs" textColor="warning" />
+                {/if}
```

Comment:
You can at least abstract this to a single component that is used in at least two places. URL / Link might even be worth componentising :shrug: Also you could use a dictionary of props per verification state to remove the if statements and easier support multiple states.